homeassistant/components/flux_led/light.py

```diff
@@ -124,8 +125,11 @@
 
 FLUX_EFFECT_LIST = sorted(list(EFFECT_MAP)) + [EFFECT_RANDOM]
 
+SERVICE_CUSTOM_EFFECT = "set_custom_effect"
+
 CUSTOM_EFFECT_SCHEMA = vol.Schema(
     {
+        vol.Required("entity_id"): str,
```

Comment:
The entity service helper will create an entity service schema for us if we just pass a dict with validators that doesn't have the entity_id item.

See the example here:
https://developers.home-assistant.io/docs/dev_101_services#entity-services

Reply:
When I remove the `entity_id` validator from the schema I get the error:

`Failed to call service flux_led/set_custom_effect. extra keys not allowed @ data['entity_id']`

That's with the schema reading:

```
CUSTOM_EFFECT_SCHEMA = vol.Schema(
    {
        vol.Required(CONF_COLORS): vol.All(
            cv.ensure_list,
            vol.Length(min=1, max=16),
            [
                vol.All(
                    vol.ExactSequence((cv.byte, cv.byte, cv.byte)), vol.Coerce(tuple)
                )
            ],
        ),
        vol.Optional(CONF_SPEED_PCT, default=50): vol.All(
            vol.Range(min=0, max=100), vol.Coerce(int)
        ),
        vol.Optional(CONF_TRANSITION, default=TRANSITION_GRADUAL): vol.All(
            cv.string, vol.In([TRANSITION_GRADUAL, TRANSITION_JUMP, TRANSITION_STROBE])
        ),
    }
)
```

Reply:
Please pass in a dictionary, not a schema.

Reply:
Don't create a `vol.Schema` instance. It should just be a dict. The devel is in the details.

Reply:
Stupid devil! OK - got it.